uktrade/data-hub-frontend
# Feature/pipeline unarchive form

## Description of change

Add a new form for unarchiving a pipeline item. This is currently hidden behind a url.

- This includes a refactor of ArchivePipelineItemFrom to create GetPipelineDetails to enable this to be shared between the two forms
- Update fixtures to include the new object properties
- Update PipelineDetails to show archive reason and date

## Test instructions

Add a company to your pipeline, edit the item and change `/edit` to `/archive` give it a reason and archive it, now click edit again and change `/edit` to `/unarchive` to see the new form. Actions can be confirmed in the inspector to see `archived: true` and then `archived: false` on the pipeline item

## Screenshots

![unarchive_form_2](https://user-images.githubusercontent.com/1481883/85039903-1be2cc80-b180-11ea-8521-d859bcb97744.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
